src/features/pass-survey/ui/ActivityItem.tsx

```diff
@@ -81,12 +81,20 @@ function ActivityItem({
   const releaseScrolling = () => setScrollEnabled(true);
 
   const onTimeIsUp = useCallback(() => {
-    next(true);
+    next({ isForced: true, shouldAutoSubmit: true });
   }, [next]);
 
   function moveToNextItem() {
-    if (!pipelineItem.additionalText?.required) {
-      setImmediate(() => next(true));
+    const isRadioItem = pipelineItem.type === 'Radio';
+    const autoAdvanceEnabled = !(
+      isRadioItem && !pipelineItem.payload.autoAdvance
+    );
+    const shouldAutoSubmit = !isRadioItem;
+
+    if (!pipelineItem.additionalText?.required && autoAdvanceEnabled) {
```

Comment:
Are you sure that this "!pipelineItem.additionalText?.required" check is not enough? If not, why?